### ISSUE TITLE
Refferring Link and Guest Achievement add-ons

### ISSUE THREAD

**Issue by qe8800df** [opencredit/badgeos#433]
We are using the referring link add-on along with the guest achievement add-on. 
 
As per our requirement a user need to click the referring link and prompt a login and after logging in they will be presented with a page that they have earned the badge.
 
But how the combination of the add-ons are working is once a user clicks the link its taking the user to the achievement of type’s page and then it displays a message that you may have earned the badge, login and upon logging in the user is directed to the main page with the list of badges and when they select the particular badge they have earned they can view that they have earned the badge.
 
Therefore I wanted to ask you if you know of any work around to match our requirements or its just how the plugins functions, since there are not enough documents available online on how to use the add-ons or customize the add-ons.

**Comment by tw2113** [comment]
hi qe8800df, sorry about the brief delay in response.

Going based on what you've said above, it sounds like the combination of the two is managing to add extra steps/clicks in between the clicking of the referring link and being logged in? It's also putting them on an unintended page once they've logged in, making them have to click to the badge that the referring link was to award, before they know they've earned it?

**Comment by qe8800df** [comment]
Yes sir that’s correct.

**Comment by tw2113** [comment]
It does seem like there is some incompatibility between the two at the moment, at least not great compatibility. Somehow, the referral link achievement is getting put into the cookie that the guest achievements addon is tracking, and its own redirect is getting run first before the referral link url gets loaded. So the referral is getting marked as awarded, just not via a direct route to the intended url.

**Comment by qe8800df** [comment]
Is there any hotfix for this issue?

**Comment by tw2113** [comment]
Nope, this is the first time it's been reported, so I don't have anything worked out to attempt to resolve it. It's mostly a case of 2 separate plugins doing their thing, and neither technically being broken overall.

**Comment by tw2113** [comment]
So I'm looking over the Referral link stuff for something unrelated, and I noticed this in the readme: "What happens if a user is not already logged in to your site when clicking a Referring Link? With the bonus "Guest Achievement" add-on activated, users will be awarded the achievement upon logging in or joining your site."

So it sounds like it is working still, and like expected, but it's just a matter of which location your user lands on that you're not liking as much?

**Comment by qe8800df** [comment]
Yes that’s correct we would like the users to land the badge awarded page immediately after logging in.

**Comment by tw2113** [comment]
I can't promise anything other than we'll look into what we can do regarding it, if anything. BadgeOS needs to keep in mind that there may also be users who like it as is too. For the most part, it's mostly a question of which plugin's redirects get encountered first in the execution of all of the code going on.
